packages/gatsby/src/query/query-runner.js

```diff
@@ -25,7 +25,28 @@ type QueryJob = {
 module.exports = async (graphqlRunner, queryJob: QueryJob) => {
   const { program } = store.getState()
 
-  const graphql = (query, context) => graphqlRunner.query(query, context)
+  const graphql = (query, context) => {
+    // Check if query takes too long, print out warning
+    const promise = graphqlRunner.query(query, context)
+    let isPending = true
+
+    const timeoutId = setTimeout(() => {
+      if (isPending) {
+        const { pages } = store.getState()
+        const page = pages.get(queryJob.id)
+        console.warn(
+          `Query in the ${page.internalComponentName} component takes too long`
```

Comment:
Query runner also executes [static queries](https://www.gatsbyjs.org/docs/static-query/) (not just page queries). Maybe we can use `queryJob.componentPath` here instead? It should be universal for both page and static queries.

Reply:
Yeah, I think that `queryJob.componentPath` looks more suitable. Thanks.